client-js/app/scripts/app.ts

```diff
@@ -351,34 +334,30 @@ window.addEventListener('WebComponentsReady', () => {
       this.plot.addData(data);
     }
 
-    stop(): Promise<void> {
+    async stop(): Promise<void> {
```

Comment:
A slight tweak on what you wrote: "Make this function return a `Promise` _which will eventually resolve_ to whatever this function (eventually) returns, or will be rejected if the function raises an exception." And of course, `async` also plumbs things together so that you can use `await` to deal with other asynchronous operation inside the function without turning the code inside out to write callbacks.

`async`/`await` here is very similar to twisted's `@inlineCallbacks`/`yield` in python. In fact, under the hood async functions are implemented with ES6 generators, so the analogy is really very close. One difference is that `async` functions can use plain old `return` to return values, whereas `@inlineCallbacks` functions in python cannot (at least until python 3).